cpp/src/Slice/Grammar.y

```diff
@@ -313,91 +307,58 @@ definitions
 // ----------------------------------------------------------------------
 definition
 // ----------------------------------------------------------------------
-: module_def
+: module_def opt_semicolon
 {
     assert($1 == nullptr || dynamic_pointer_cast<Module>($1));
 }
-opt_semicolon
-| class_decl
+| class_decl ';'
 {
     assert($1 == nullptr || dynamic_pointer_cast<ClassDecl>($1));
 }
-';'
-| class_decl
-{
-    currentUnit->error("';' missing after class forward declaration");
-}
```

Comment:
I removed all these custom rules for detecting missing semicolons.
This made sense before, when all you'd see was `syntax error`.

But now that Bison tells you it expected `';'`, I don't think these extra rules add anything.